services/workflows-service/src/customer/customer.controller.external.intg.test.ts

```diff
@@ -0,0 +1,205 @@
+import { CustomerRepository } from './customer.repository';
+import { EndUserService } from '@/end-user/end-user.service';
+import { before } from 'lodash';
+import { AllExceptionsFilter } from '@/common/filters/AllExceptions.filter';
+import { string } from 'zod';
+
+describe('#CustomerControllerExternal', () => {
+  let app: INestApplication;
+  let customerService: CustomerService;
+  let project: Project;
+  let customer: Customer;
+  beforeAll(cleanupDatabase);
+  afterEach(tearDownDatabase);
+
+  beforeAll(async () => {
+    // await cleanupDatabase();
+
+    const servicesProviders = [
+      AllExceptionsFilter,
+      CustomerService,
+      CustomerRepository,
+      EndUserService,
+      EndUserRepository,
+      EntityRepository,
+      FilterService,
+      ProjectScopeService,
+      FilterRepository,
+      FileRepository,
+      FileService,
+      StorageService,
+      WorkflowEventEmitterService,
+      BusinessRepository,
+      WorkflowDefinitionRepository,
+      WorkflowRuntimeDataRepository,
+      WorkflowService,
+      EventEmitter2,
+      PrismaService,
+      UserService,
+      UserRepository,
+      SalesforceService,
+      SalesforceIntegrationRepository,
+      PasswordService,
+      WorkflowTokenService,
+      WorkflowTokenRepository,
+      WorkflowRuntimeDataRepository,
+      UiDefinitionRepository,
+      UiDefinitionService,
+    ];
+    customerService = (await fetchServiceFromModule(CustomerService, servicesProviders, [
+      PrismaModule,
+    ])) as unknown as CustomerService;
+    app = await initiateNestApp(
+      app,
+      [
+        {
+          provide: CustomerService,
+          useValue: customerService,
+        },
+        ...servicesProviders,
+      ],
+      [CustomerControllerExternal],
+      [PrismaModule, ClsModule],
+    );
+  });
+
+  describe('POST /subscriptions', () => {
+    let customerId: string;
+    let projectId: string;
+
+    beforeEach(async () => {
+      const prismaClient = app.get(PrismaService);
+      customerId = faker.datatype.uuid();
+
+      customer = await createCustomer(
+        prismaClient,
+        customerId,
+        'secret3',
+        '',
+        '',
+        'webhook-shared-secret',
+      );
+      projectId = faker.datatype.uuid();
+      project = await createProject(prismaClient, customer, projectId);
+
+      const { id, ...restCustomerFields } = await customerService.getByProjectId(projectId);
+
+      expect(restCustomerFields).toMatchInlineSnapshot(`
+      {
+        "country": "GB",
+        "displayName": "Customer ${customerId}",
+        "faviconImageUri": "",
+        "language": "en",
+        "logoImageUri": "",
+        "name": "Customer ${customerId}",
+        "projects": [
+          {
+            "customerId": "${customerId}",
+            "id": "${projectId}",
+            "name": "Project ${projectId}",
+          },
+        ],
+        "subscriptions": null,
+        "websiteUrl": null,
+      }
+    `);
+    });
```

Comment:
Not sure I follow how its easier?
```suggestion
      expect(restCustomerFields).toMatchObject({
  country: 'GB',
  displayName: 'Customer ${customerId}',
  faviconImageUri: '',
  language: 'en',
  logoImageUri: '',
  name: 'Customer ${customerId}',
  projects: [
    {
      customerId: '${customerId}',
      id: '${projectId}',
      name: 'Project ${projectId}',
    },
  ],
  subscriptions: null,
  websiteUrl: null,
});
```

Reply:
Using a real object instead of a string object is better IMO cuz you have autocompletion and have less room to mistake.